### DISCUSSION TITLE
Support for "Retrieve a Record Using an External ID"

### DISCUSSION THREAD

**Discussion by borys-kupar** [jsforce/jsforce#1487] [Ideas]
It seems like it's not possible to retrieve object by external id using jsforce. Any plans on implementing that? I could also help with implementation and providing a PR.

Documentation: https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/using_resources_retrieve_with_externalid.htm?search_text=update

**Comment by pabhishek09** [comment]
+1 . Any ideas why the PR is not merged?

**Comment by borys-kupar** [comment]
No, would love to get some feedback from @stomita

**Comment by kmanzana** [comment]
Bump on this. Will probably be using the fork for now

**Comment by cristiand391** [comment]
triaging old issues... moved to GH discussions/ideas

current workaround:
https://github.com/jsforce/jsforce/pull/820#issuecomment-1016939087

that works because internally the URL is built by `join`ing strings:
https://github.com/jsforce/jsforce/blob/b661afdd21729812974f1056e392f2f58fc21096/src/connection.ts#L883

and with `id` being `My_External_ID__c/123123121` you get the the valid URL documented here:
https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/using_resources_retrieve_with_externalid.htm?search_text=update
